### PR TITLE
Add local correlation fields for Brave

### PR DESCRIPTION
**Pull request by adispezo**
Currently with the following config example it's not possible to use BaggageField.getByName, BaggageField.updateValue for example.

```yaml
management:
    tracing:
        baggage:
            correlation:
                fields: 
                    - some_local_field
                    - some_remote_field
           remote-fields:
               - some_remote_field
```

Only the "some_remote_field" can be retrieved and updated.

With the code proposed also the "local" correlation fields will be accessible.

### CODE REVIEW

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/BraveAutoConfiguration.java [hunk 1]
```diff
@@ -244,6 +245,19 @@ BaggagePropagationCustomizer remoteFieldsBaggagePropagationCustomizer() {
 			};
 		}
 
+		@Bean
+		@Order(0)
```
Comment (mhalbritter) [inline spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/BraveAutoConfiguration.java:249]: We already have a `Order(0)` `BaggagePropagationCustomizer`, i'd prefer `Order(1)`.

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/BraveAutoConfiguration.java [hunk 2]
```diff
@@ -244,6 +245,19 @@ BaggagePropagationCustomizer remoteFieldsBaggagePropagationCustomizer() {
...
+		BaggagePropagationCustomizer localFieldsBaggagePropagationCustomizer() {
+			return (builder) -> {
+				final List<String> localFields = new ArrayList<>(
```
Comment (mhalbritter) [inline spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/BraveAutoConfiguration.java:252]: I don't think it's a good idea to use the `getFields()` for both remote and local fields. In https://github.com/spring-projects/spring-boot/issues/33991 we want to add tagged fields. I'd prefer an explicit configuration property for that, as we have it for the remote fields.

**Review by mhalbritter — CHANGES_REQUESTED** [review]

**Comment by mhalbritter** [comment]
Hello,

thanks for the PR! I've added some comments.

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/BraveAutoConfiguration.java (hunk 1 above)
Comment (adispezo) [reply spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/BraveAutoConfiguration.java]: I have changed the order.

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/BraveAutoConfiguration.java (hunk 2 above)
Comment (adispezo) [reply spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/BraveAutoConfiguration.java]: I have missed that conversation for some reason, but I have implemented both local and tagged fields. I also think it is to have the extra property fields.

**Comment by mhalbritter** [comment]
Hey @adispezo , thanks for the changes! I'll get some feedback from the observability team on the local field change in Brave and then come back to you.

**Comment by mhalbritter** [comment]
Looks good, thank you!

Note to the boot team: when merging, we need something like this for OpenTelemetry, too.

**Comment by marcingrzejszczak** [comment]
There's no way to do it easily with OTel. What currently is being done is that all baggage gets extracted / injected (e.g.https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/baggage/propagation/W3CBaggagePropagator.java#L50). We would need to wrap their baggage propagation mechanism into our one that would modify the current baggage not to contain local keys and then delegate to their code.

**Comment by mhalbritter** [comment]
Hm :( In this case, we won't support local fields when using OpenTelemetry. We could issue a log WARN statement if local fields are configured and OTel is used

**Comment by mhalbritter** [comment]
I have some polished changes and tagged fields for Otel here: https://github.com/mhalbritter/spring-boot/tree/pr/37435

**Comment by mhalbritter** [comment]
Before merging this, https://github.com/spring-projects/spring-boot/issues/38724 needs to be considered.

**Comment by mhalbritter** [comment]
Merged, thank you. To accommodate the changes from https://github.com/spring-projects/spring-boot/issues/38724, i had to refactor a bit.

**Comment by o-shevchenko** [comment]
Hi
Do we already support tags `tag-fields` in Spring Boot 3.3.0? 
We've set `tag-fields`, but we still don't see tags sent to the OTEL collector.

**Comment by mhalbritter** [comment]
Yes, this should work. If it doesn't, please open a new issue with a small reproducer. Thanks!

**Comment by o-shevchenko** [comment]
Thanks @mhalbritter !
Should it be supported for all tracers or only Brave? I don't see it works with `io.micrometer:micrometer-tracing-bridge-otel`

**Comment by mhalbritter** [comment]
Works with both tracers. [See the Otel configuration here](https://github.com/spring-projects/spring-boot/commit/25517350938f843f3077aa74d883d9525d1850af#diff-ffad1b43e8b173c888b93cf9becac2d848a524b9db2f5c30ba387e1b80a59e5cR145).

**Pull request closed without merging**